go/libraries/doltcore/env/environment.go

```diff
@@ -158,6 +174,15 @@ func Load(ctx context.Context, hdp HomeDirProvider, fs filesys.Filesys, urlStr,
 		}
 	}
 
+	if dbLoadErr == nil {
+		postCommitHooks, dbLoadErr := initializeCommitHooks(ctx, dEnv)
+		if dbLoadErr != nil {
+			dEnv.DBLoadError = dbLoadErr
+		} else {
+			dEnv.DoltDB = ddb.WithCommitHooks(ctx, postCommitHooks)
```

Comment:
This is phrased badly: you are using a With method on a different variable than the one you are assigning it to. They point to the same object in this case, but that could change and then this code would break. Use:

```go
dEnv.DoltDB = dEnv.DoltDB.WithCommitHooks(...)
```